src/App.tsx

```diff
@@ -1,15 +1,52 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [filterValue, setFilterValue] = useState('');
+  const [debouncedFilterValue, setDebouncedFilterValue] = useState(filterValue);
+
+  const debouncedSetFilterValue = useCallback(
+    debounce((value: string) => {
+      setDebouncedFilterValue(value);
+    }, 300),
+    [],
+  );
+
+  const handleSelectedPerson = (person: Person | null) => {
```

Comment:
Follow naming conventions for methods. The method name `handleSelectedPerson` should reflect that it's an event handler. A better name would be `handlePersonSelect` to conform with the conventions.